workflow/controller/operator.go

```diff
@@ -2829,17 +2829,16 @@ func (woc *wfOperationCtx) loadExecutionSpec() (wfv1.TemplateReferenceHolder, wf
 		}
 	}
 
-	woc.wfSpec = woc.wf.Status.StoredWorkflowSpec
-
-	entrypoint := woc.wf.Spec.Entrypoint
-	if entrypoint == "" {
-		entrypoint = woc.wfSpec.Entrypoint
+	// Merge the workflow spec and storedWorkflowspec.
+	targetWf := wfv1.Workflow{Spec: *woc.wf.Status.StoredWorkflowSpec}
+	err := wfutil.MergeTo(woc.wf, &targetWf, true)
```

Comment:
if you flip the two arguments, you overwrite can be false? it would then always be false and can be removed

Reply:
True. The flag will be good idea so the caller can control expected behavior otherwise caller always needs to manipulate arguments to get the desire target.